### DISCUSSION TITLE
Streaming endpoint similarly to streamed response from http client requests

### DISCUSSION THREAD

**Discussion by brynjen** [serverpod/serverpod#1450] [Q&A] [answered]
Currently there is no way to stream a serialized response similar to what you can do in dart with http package using a request where you get a streamed response. You can on the other hand achieve similar result using in serverpod with a StreamingSession and handleStreamMessage, but it requires that you authenticate user first.

Ideally you would be able to have an endpoint that supports something similar to

**Future<Stream<Answer>> streamAnswer(Session session, String question) {}**

Alternatives would of course be to authenticate user and use websockets/StreamingSession as it seems to be only way to stream.

In these GPT times where data could take a long time to generate, streaming responses would be handy without having to necessrily authenticate and use the websockets since it is already supported in the http package from post/get requests.

**Comment by vlidholt** [comment, marked as answer]
There is no need to authenticate the user when using streaming sessions, just do not set the `requireLogin` property on your Endpoint.

An example is Pixorama (comes with full source code):
https://pixorama.live/

**Reply by brynjen** [reply]
I must have missed that - and considering I've tested pixorama and looked through the endpoints code that was not my best moment 😅 Thanks for the speedy and informative response though 🙂

**Reply by brynjen** [reply]
hmm...finally had time to sit down with the code again. I currently don't override that property. And it is false by default.

I get the streamOpened and streamClosed on the server, but nothing in the handleStreamMessage when I sendStreamMessage from client after connection is established. Still working on solving that issue, but no luck so far. Will try to clone pixorama and see if I can debug it from there
